### PR TITLE
Add support for fake tickers

### PR DESCRIPTION
**Pull request by tflach**
I spend a little time on extending your library to add support for fake tickers.
Currently this requires surrounding the Advance() calls with BlockUntil() to make sure that the ticker can go to sleep before Advance() is called and afterwards wake up before we start consuming ticks.
Let me know what you think.

### CODE REVIEW

**Comment by tflach** [comment]
Hello Jon,
did you have a chance to take a look at this?

File: clockwork.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ type Clock interface {
 	Sleep(d time.Duration)
 	Now() time.Time
 	Since(t time.Time) time.Duration
+	NewTicker(d time.Duration) Ticker
```
Comment (pierrebeaucamp) [inline clockwork.go]: This is technically not following the exact definition of `NewTicker` from the stdlib. There it is `func NewTicker(d Duration) *Ticker`. But I don't know if this is a concern

**Comment by pierrebeaucamp** [comment]
I think that the `BlockUntil()` requirement could be resolved by #12 
Also, I couldn't help but notice that the standard library implements tickers using timers, which might get added in #13.

**Comment by jonboulle** [comment]
LGTM. Sorry about the horrendous delay.

**Pull request merged**